jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3VirtualBucket.kt

```diff
@@ -19,14 +19,13 @@ import software.aws.toolkits.jetbrains.services.s3.upload
 import java.io.InputStream
 import java.io.OutputStream
 
-class S3VirtualBucket(val s3Bucket: Bucket, val client: S3Client) : LightVirtualFile() {
-    override fun getName(): String = s3Bucket.name()
+class S3VirtualBucket(val s3Bucket: Bucket, val client: S3Client) : LightVirtualFile(s3Bucket.name()) {
     override fun isWritable(): Boolean = false
     override fun getPath(): String = s3Bucket.name()
     override fun isValid(): Boolean = true
     override fun getParent(): VirtualFile? = null
     override fun toString(): String = s3Bucket.name()
-    override fun isDirectory(): Boolean = true
+    override fun isDirectory(): Boolean = false /* Unit tests refuse to open this in an editor if this is true */
```

Comment:
that's kinda weird.

Reply:
Ya, I did a lot of step through debugging to find these checks :/